sklearn/linear_model/quantile.py

```diff
@@ -0,0 +1,314 @@
+
+    .. [2] Chen, C., & Wei, Y. (2005).
+           Computational issues for quantile regression.
+           Sankhya: The Indian Journal of Statistics, 399-417.
+    """
+
+    def __init__(self, quantile=0.5,
+                 max_iter=10000, alpha=0.0001, l1_ratio=0.0,
+                 warm_start=False, fit_intercept=True,
+                 normalize=False, copy_X=True,
+                 gamma=1e-2, gtol=1e-4, xtol=1e-6,
+                 gamma_decrease=0.1, n_gamma_decreases=100):
+        self.quantile = quantile
+        self.max_iter = max_iter
+        self.alpha = alpha
+        self.l1_ratio = l1_ratio
+        self.warm_start = warm_start
+        self.fit_intercept = fit_intercept
+        self.copy_X = copy_X
+        self.normalize = normalize
+        self.gtol = gtol
+        self.xtol = xtol
+        self.gamma = gamma
+        self.gamma_decrease = gamma_decrease
+        self.n_gamma_decreases = n_gamma_decreases
+
+    def fit(self, X, y, sample_weight=None):
+        """Fit the model according to the given training data.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Training vector, where n_samples in the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape (n_samples,)
+            Target vector relative to X.
+
+        sample_weight : array-like, shape (n_samples,)
+            Weight given to each sample.
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        X, y = check_X_y(
+            X, y, copy=False, accept_sparse=['csr'], y_numeric=True)
+
+        X, y, X_offset, y_offset, X_scale = self._preprocess_data(
+            X, y, self.fit_intercept, self.normalize, self.copy_X,
+            sample_weight=sample_weight)
+
+        if sample_weight is not None:
+            sample_weight = np.array(sample_weight)
+            check_consistent_length(y, sample_weight)
+        else:
+            sample_weight = np.ones_like(y)
+
+        if self.quantile >= 1.0 or self.quantile <= 0.0:
+            raise ValueError(
+                "Quantile should be strictly between 0.0 and 1.0, got %f"
+                % self.quantile)
+
+        if self.warm_start and hasattr(self, 'coef_'):
+            parameters = np.concatenate(
+                (self.coef_, [self.intercept_]))
+        else:
+            if self.fit_intercept:
+                parameters = np.zeros(X.shape[1] + 1)
+            else:
+                parameters = np.zeros(X.shape[1] + 0)
+
+        # solve sequence of optimization problems
+        # with different smoothing parameter
+        total_iter = []
+        loss_args = (X, y, self.quantile, self.alpha, self.l1_ratio,
+                     self.fit_intercept, sample_weight)
+        for i in range(self.n_gamma_decreases):
+            gamma = self.gamma * self.gamma_decrease ** i
```

Comment:
So if `l1_ratio=0.0` the penalty is smooth, do we actually need to do this loop?  On failure to converge I think this would try anther round of LBFGS optimization while it should just exist with a ConvergenceWarning in that case I think?

Reply:
Nevermind, I understand it's not smooth even without l1_ratio..